iree/hal/api.cc

```diff
@@ -522,7 +537,10 @@ static iree_status_t iree_hal_format_buffer_elements_recursive(
     iree_device_size_t dim_stride =
         dim_length * iree_hal_element_byte_count(element_type);
     if (data.data_length < dim_stride * shape[0]) {
-      return IREE_STATUS_OUT_OF_RANGE;
+      return iree_make_status(
+          IREE_STATUS_OUT_OF_RANGE,
+          "input data underflow: data_length=%zu < expected=%zu",
+          data.data_length, dim_stride * shape[0]);
```

Comment:
I'm seeing warnings about printf type mismatches here (depending on the toolchain used to compile?):

```
iree/hal/api.cc:540:29: error: format specifies type 'size_t' (aka 'unsigned int') but the argument has type 'unsigned long long' [-Werror,-Wformat]
          data.data_length, dim_stride * shape[0]);
                            ^~~~~~~~~~~~~~~~~~~~~
```

Can you add a cast, or adjust the types used? You're mixing `size_t iree_host_size_t` and `uint64_t iree_device_size_t`.
```suggestion
          "input data underflow: data_length=%zu < expected=%zu",
          data.data_length, static_cast<size_t>(dim_stride * shape[0]));
```